sockeye/inference.py

```diff
@@ -1132,7 +1133,10 @@ def translate(self, trans_inputs: List[TranslatorInput]) -> List[TranslatorOutpu
             if rest > 0:
                 logger.debug("Extending the last batch to the full batch size (%d)", self.batch_size)
                 batch = batch + [batch[0]] * rest
-            batch_translations = self._translate_nd(*self._get_inference_input(batch))
+            max_output_lengths = []
+            for i in range(len(batch)):
+                max_output_lengths.append(self.models[0].get_max_output_length(len(batch[i])))
+            batch_translations = self._translate_nd(*self._get_inference_input(batch), max_output_lengths)
```

Comment:
how about adding the creation of `max_output_lengths` directly to `_get_inference_input`? Also I'd directly create the NDArray and pass that instead of the list to _beam_search.